tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -23,3 +25,48 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.sql'
         client = 'SqlManagementClient'
         enum_spec = ('servers', 'list', None)
+
+
+@SqlServer.filter_registry.register('firewall')
+class SqlServerFirewallFilter(ValueFilter):
+    """Filters SQL servers by the firewall rules
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: servers-without-firewall
+                resource: azure.sqlserver
+                filters:
+                  - type: firewall
+                    key: firewall_rules
+                    value_type: size
+                    op: eq
+                    value: 0
+    """
+
+    schema = type_schema('firewall', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+        client = self.manager.get_client()
+
+        def _query_firewall_rules(resource):
+            query = client.firewall_rules.list_by_server(
+                resource['resourceGroup'],
+                resource['name'])
+
+            rules = [
+                {
+                    'name': r.name,
+                    'start_ip_address': r.start_ip_address,
+                    'end_ip_address': r.end_ip_address
+                }
+                for r in query]
+
+            resource['firewall_rules'] = rules
```

Comment:
Per Kapil's feedback in another PR, it is better to add `c7n:` prefix annotations, e.g. `c7n:firewall_rules`